extensions/wikia/IndexingPipeline/PipelineConnectionBase.class.php

```diff
@@ -61,92 +42,33 @@ public function publish( $routingKey, $body ) {
 		);
 	}
 
-	/**
-	 * Will ack on all true and null results, on false no action, on error no_ack
-	 * @param AMQPMessage $msg
-	 */
-	protected function route( AMQPMessage $msg ) {
-		try {
-			$result = call_user_func( self::$binding[ 0 ], json_decode( $msg->body ) );
-		} catch ( Exception $e ) {
-			$msg->delivery_info[ 'channel' ]->basic_reject( $msg->delivery_info[ 'delivery_tag' ], self::NO_REQUEUE );
-		}
-		if ( !isset( $result ) || $result ) {
-			$msg->delivery_info[ 'channel' ]->basic_ack( $msg->delivery_info[ 'delivery_tag' ] );
-		}
-	}
-
 	/** @return \PhpAmqpLib\Channel\AMQPChannel */
-	protected function setUpPublishChannel() {
-		if ( !isset( $this->publishChannel ) || !$this->publishChannel->is_open ) {
-			$pubConn = $this->getPublishConnection();
-			$this->publishChannel = $pubConn->channel();
+	protected function getChannel() {
+		if ( !isset( $this->channel ) || !$this->channel->is_open ) {
+			$pubConn = $this->getConnection();
+			$this->channel = $pubConn->channel();
 		}
-		return $this->publishChannel;
-	}
 
-	/** @return \PhpAmqpLib\Channel\AMQPChannel */
-	protected function setUpMainChannel() {
-		// check if binding was set
-		if ( is_array( self::$binding ) && is_callable( self::$binding[ 0 ] ) ) {
-			if ( !isset( $this->mainChannel ) ) {
-				$mainConn = $this->getMainConnection();
-				$this->mainChannel = $mainConn->channel();
-				$qName = $this->getQueueName();
-				$this->mainChannel->queue_declare( $qName, false, true, false, true, false, [
-					'x-dead-letter-exchange' => [ 'S', $this->deadExchange ]
-				] );
-				$this->mainChannel->queue_bind( $qName, $this->exchange, key( self::$binding ) );
-				$this->mainChannel->basic_qos( self::$prefetch );
-				$this->mainChannel->basic_consume( $qName, '', false, false, false, false, [ $this, 'route' ] );
-			}
-			return $this->mainChannel;
-		}
-		throw new WikiaException( 'Binding must be set in child class' );
+		return $this->channel;
 	}
 
-	protected function closeMainChannel() {
-		if ( isset( $this->mainChannel ) ) {
-			$this->mainChannel->close();
-			unset( $this->mainChannel );
-		}
-	}
-
-	protected function getQueueName() {
-		return implode( '.', [ get_class( $this ), key( self::$binding ), self::QUEUE_NAME_POSTFIX ] );
-	}
-
-	protected function createNewConnection() {
+	protected function createConnection() {
 		return new AMQPConnection( $this->host, $this->port, $this->user, $this->pass, $this->vhost );
```

Comment:
it should be caught and handled here as far as I see:
https://github.com/Wikia/app/pull/8862/files#diff-51080aaef54b01dd7a86c59c827a2761R170